File: getAccess.py
import time
import json
import hmac
import hashlib
import requests

shop_id = 37872
partner_id = 1005366
partner_key = "170ddadd5bd5ae843c7ce110a883066061a12f2fa5384091edcd16d202c4a02b"

def get_access_token_shop_level(shop_id, partner_id, partner_key, refresh_token):
    timest = int(time.time())
    host = "https://partner.shopeemobile.com"
    path = "/api/v2/auth/access_token/get"
    body = {"shop_id": shop_id, "refresh_token": refresh_token}
    base_string = "%s%s%s"%(partner_id, path, timest)
    sign = hmac.new(partner_key, base_string, hashlib.sha256).hexdigest()
    url = host + path + "?partner_id=%s&timestamp=%s&sign=%s"%(partner_id, timest, sign)
    
    headers = {"Content-Type": "application/json"}
    resp = requests.post(url, json=body, headers=headers)

    ret = json.loads(resp.content)
    access_token = ret.get("access_token")
    new_refresh_token = ret.get("refresh_token")
    return access_token, new_refresh_token

def get_access_token_merchant_level(merchant_id, partner_id, partner_key, refresh_token):
    timest = int(time.time())
    host = "https://partner.shopeemobile.com"
    path = "/api/v2/auth/access_token/get"
    body = {"merchant_id": merchant_id, "refresh_token": refresh_token}
    base_string = "%s%s%s"%(partner_id, path, timest)
    sign = hmac.new(partner_key, base_string, hashlib.sha256).hexdigest()
    url = host + path + "?partner_id=%s&timestamp=%s&sign=%s"%(partner_id, timest, sign)

    headers = {"Content-Type": "application/json"}
    resp = requests.post(url, data=json, headers=headers)

    ret = json.loads(resp.content)
    access_token = ret.get("access_token")
    new_refresh_token = ret.get("refresh_token")
    return access_token, new_refresh_token

File: getAccess.js
const express = require("express");
const axios = require("axios").default;
const cors = require("cors");
const app = express();
const { createHmac } = require("crypto");

const fs = require("fs");
const http = require("http");
const https = require("https");
http.createServer(app).listen(3000);
// const optionHttps = {
//     ca: fs.readFileSync("/etc/letsencrypt/live/suml.xyz/fullchain.pem"),
//     key: fs.readFileSync("/etc/letsencrypt/live/suml.xyz/privkey.pem"),
//     cert: fs.readFileSync("/etc/letsencrypt/live/suml.xyz/cert.pem"),
// };
// https.createServer(optionHttps, app).listen(443);
const host = "https://partner.test-stable.shopeemobile.com";
const path = "/api/v2/auth/token/get";

app.use(cors({ origin: true, credentials: true }));
app.use(express.urlencoded({ extended: false }));
app.use(express.json());

const timestamp = Math.round(Date.now() / 1000);
// const timestamp = 1641527426;
const code = "6e596567485a635a7244616f4f745762";
const partner_id = 1000639;
const partner_key =
    "90e0bd35c8a63d24052d2acb14ee4ae98e2f7a76e7428c3fa6e9da941960cac0";
const baseString = `${partner_id}${path}${timestamp}`;
const sign = createHmac("sha256", partner_key).update(baseString).digest("hex");
const shop_id = 37698;

const params = {
    partner_id,
    sign,
    timestamp,
};
const data = {
    code,
    partner_id,
    shop_id,
};
const options = {
    method: "POST",
    url: `${host}${path}`,
    params,
    data,
};
axios
    .request(options)
    .then((response) => {
        console.log("response", response.data);
    })
    .catch((error) => {
        console.error("generate error", error.response);
    });

module.exports = app;

// response {
//     access_token: '6e6671794c5761576367584858435972',
//     error: '',
//     request_id: '2d1bf5c38b03a86cd00c5936ea296875',
//     message: '',
//     expire_in: 14379,
//     refresh_token: '4548694e5a786353766d554a6d636f42'
